### PR TITLE
fix: Return errors from typeJoinOne

### LINKED ISSUE #1711: typeJoinOne silently discards errors in `Next`
`valuesPrimary` and `valuesSecondary` do this, and there appears to be no good reason for it (likely historical)

### PR DESCRIPTION
**Pull request by AndrewSisley**
## Relevant issue(s)

Resolves #1711

## Description

Returns errors from typeJoinOne instead of ignoring them.

I assume this is a historical accident.

### CODE REVIEW

File: planner/type_join.go [hunk 1]
```diff
@@ -367,32 +373,31 @@ func (n *typeJoinOne) valuesSecondary(doc core.Doc) core.Doc {
...
 	if err != nil {
-		return core.Doc{}
+		return core.Doc{}, err
 	}
 
 	// We have to reset the scan node after appending the new key-filter
 	if err := n.subType.Init(); err != nil {
-		log.ErrorE(n.p.ctx, "Sub-type initialization error at scan node reset", err)
-		return doc
+		return doc, errors.Wrap("sub-type initialization error at scan node reset", err)
```
Comment (fredcarle) [inline planner/type_join.go:381]: todo: Add to `errors.go`

File: planner/type_join.go [hunk 2]
```diff
@@ -408,8 +413,7 @@ func (n *typeJoinOne) valuesPrimary(doc core.Doc) core.Doc {
 
 	// re-initialize the sub type plan
 	if err := n.subType.Init(); err != nil {
-		log.ErrorE(n.p.ctx, "Sub-type initialization error at scan node reset", err)
-		return doc
+		return doc, errors.Wrap("sub-type initialization error at scan node reset", err)
```
Comment (fredcarle) [inline planner/type_join.go:416]: todo: After adding the above to `errors.go`, you can use the defined error function here :)

File: planner/type_join.go (hunk 1 above)
Comment (AndrewSisley) [reply planner/type_join.go]: sorted

File: planner/type_join.go (hunk 2 above)
Comment (AndrewSisley) [reply planner/type_join.go]: sorted

**Pull request merged**